### PR TITLE
MM-20542: Migrate 'components/user_settings/display/user_settings_theme' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR 
- migrates 'components/user_settings/display/user_settings_theme' module, reorganize it in dedicated folders and associated tests to TypeScript
- fixes a bug when a color is selected from color picker, first selection doesn't work. User needs to click two times

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13699
JIRA: https://mattermost.atlassian.net/browse/MM-20542

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by sinansonmez** [comment]
The status is draft because of a bug I need to fix. When you choose a color first time in color chooser, you need to select 2 times. After fixing that issue, it will be ready.

**Comment by sinansonmez** [comment]
/update-branch to receive latest version of the master without React 17

File: components/color_input.tsx [hunk 1]
```diff
@@ -71,6 +71,7 @@ export default class ColorInput extends React.PureComponent<Props, State> {
     };
 
     public handleColorChange = (newColorData: ColorResult) => {
+        this.setState({focused: false});
```
Comment (sinansonmez) [inline components/color_input.tsx:74]: This is related to a bug mention in PR description

**Comment by sinansonmez** [comment]
Kindly pinging for review @M-ZubairAhmed and @AshishDhama

**Review by AshishDhama — APPROVED** [review]

**Comment by AshishDhama** [comment]
@sinansonmez Thanks for your patience, Keep up the good work.

File: components/user_settings/display/user_settings_theme/custom_theme_chooser/custom_theme_chooser.tsx [hunk 2]
```diff
@@ -319,7 +347,6 @@ export default class CustomThemeChooser extends React.PureComponent {
                             <select
                                 id='codeThemeSelect'
                                 className='form-control'
-                                type='text'
```
Comment (M-ZubairAhmed) [inline components/user_settings/display/user_settings_theme/custom_theme_chooser/custom_theme_chooser.tsx:322]: 🎉

File: components/user_settings/display/user_settings_theme/custom_theme_chooser/custom_theme_chooser.tsx [hunk 3]
```diff
@@ -328,7 +355,7 @@ export default class CustomThemeChooser extends React.PureComponent {
                             <OverlayTrigger
                                 placement='top'
                                 overlay={popoverContent}
-                                ref='headerOverlay'
+                                ref={this.headerOverlayRef}
```
Comment (M-ZubairAhmed) [inline components/user_settings/display/user_settings_theme/custom_theme_chooser/custom_theme_chooser.tsx:358]: thanks for migrating away from string refs

File: components/user_settings/display/user_settings_theme/user_settings_theme.tsx [hunk 4]
```diff
@@ -1,61 +1,77 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-
 import React from 'react';
```
Comment (M-ZubairAhmed) [inline components/user_settings/display/user_settings_theme/user_settings_theme.tsx:5]: nit : no line space

File: components/user_settings/display/user_settings_theme/user_settings_theme.tsx [hunk 5]
```diff
@@ -1,61 +1,77 @@
...
 import SettingItemMax from 'components/setting_item_max.jsx';
 import SettingItemMin from 'components/setting_item_min';
-import ImportThemeModal from 'components/user_settings/import_theme_modal.tsx';
+import ImportThemeModal from 'components/user_settings/import_theme_modal';
 
 import {Constants, ModalIdentifiers} from 'utils/constants';
-import * as Utils from 'utils/utils';
+import {applyTheme} from 'utils/utils';
+
+import {Theme} from 'mattermost-redux/types/themes';
```
Comment (M-ZubairAhmed) [inline components/user_settings/display/user_settings_theme/user_settings_theme.tsx:15]: nit : move up after lib imports

File: components/user_settings/display/user_settings_theme/user_settings_theme.tsx [hunk 6]
```diff
@@ -1,61 +1,77 @@
...
 import {FormattedMessage} from 'react-intl';
 
 import SettingItemMax from 'components/setting_item_max.jsx';
 import SettingItemMin from 'components/setting_item_min';
-import ImportThemeModal from 'components/user_settings/import_theme_modal.tsx';
+import ImportThemeModal from 'components/user_settings/import_theme_modal';
 
 import {Constants, ModalIdentifiers} from 'utils/constants';
-import * as Utils from 'utils/utils';
+import {applyTheme} from 'utils/utils';
```
Comment (M-ZubairAhmed) [inline components/user_settings/display/user_settings_theme/user_settings_theme.tsx:13]: kudus for selective imports 🎉

**Review by M-ZubairAhmed — APPROVED** [review]
🆗  It looks good to me. Thank you

**Comment by furqanmlk** [comment]
e2e-test

**Review by furqanmlk — APPROVED** [review]

**Comment by sinansonmez** [comment]
@furqanmlk thanks for the approval. Could someone merge the PR?

**Comment by sinansonmez** [comment]
Kindly pinging to merge the PR @AshishDhama @M-ZubairAhmed @furqanmlk

**Pull request merged**